src/parsing.js

```diff
@@ -102,10 +102,83 @@
     return matchedUnits;
   }
 
+  function ignore(value) {
+    return function(input) {
+      var result = value(input);
+      if (result)
+        result[0] = undefined;
+      return result;
+    }
+  }
+
+  // Regex should be anchored with /^
+  function consumeToken(regex, string) {
+    var result = regex.exec(string);
+    if (result) {
+      result = regex.ignoreCase ? result[0].toLowerCase() : result[0];
+      return [result, string.substr(result.length)];
+    }
+  }
+
+  function consumeNumber(string) {
+    var result = consumeToken(/^[0-9]?\.?[0-9]+/, string);
+    if (result && result[0]) {
+      result[0] = parseFloat(result[0]);
+    }
+    return result;
+  }
+
+  function consumeTrimmed(consumer, string) {
+    string = string.replace(/^\s*/, '');
+    var result = consumer(string);
+    if (result) {
+      return [result[0], result[1].replace(/^\s*/, '')];
```

Comment:
Nit
`if (result) {
    result[1] = result[1].replace(/^\s*/, '');
}
return result;
`
to match consumeNumber?